exporter/src/main/java/io/zeebe/exporters/kafka/KafkaExporter.java

```diff
@@ -100,42 +100,58 @@ public void open(final Controller controller) {
 
   @Override
   public void close() {
-    closeInternal();
+    if (isClosed) {
+      return;
+    }
+
+    isClosed = true;
+    recordBatch.cancel();
     checkCompletedInFlightRequests();
-    requests.cancelAll();
+    recordBatch.clear();
+    closeProducer();
 
     logger.debug("Closed exporter {}", id);
   }
 
-  @SuppressWarnings("rawtypes")
   @Override
   public void export(final Record record) {
-    // The producer may be closed prematurely if an unrecoverable exception occurred, at which point
-    // we ignore any further records; this way we do not block the exporter processor, and on
-    // restart will reprocess all other records that we "missed" here.
-    if (producer == null) {
-      requests.cancelAll();
-      logger.debug("Exporter {} was prematurely closed earlier; skipping record {}", id, record);
+    if (isClosed) {
+      logger.warn("Expected to export {}, but the exporter is already closed", record);
       return;
     }
 
-    if (recordHandler.test(record)) {
-      final ProducerRecord<RecordId, Record> kafkaRecord = recordHandler.transform(record);
-      final Future<RecordMetadata> future = producer.send(kafkaRecord);
-      final Request request = new Request(record.getPosition(), future);
+    if (!recordHandler.test(record)) {
+      logger.trace("Ignoring record {}", record);
+      return;
+    }
 
-      while (!requests.offer(request)) {
-        logger.trace("Too many in flight records, blocking until at least one completes...");
-        requests.consume(this::updatePosition);
-      }
+    if (!recordHandler.test(record)) {
+      logger.trace("Ignoring record {}", record);
+      return;
+    }
```

Comment:
Is it necessary to do this twice ?

Reply:
Nope, that's a leftover from refactoring :see_no_evil: